plugins/com.aptana.ui/src/com/aptana/ui/dialogs/MultipleInputMessageDialog.java

```diff
@@ -206,13 +206,29 @@ else if (element instanceof JsonNode)
 						return element.toString();
 					}
 				});
-				combo.setSelection(new StructuredSelection(values.get(0)));
+				JsonNode defaultValue = question.path(DEFAULT);
+				if (defaultValue != null && values instanceof ArrayNode)
+				{
+					ArrayNode arrayNode = (ArrayNode) values;
+					for (JsonNode node : arrayNode)
+					{
+						if (ObjectUtil.areEqual(node.get(VALUE).textValue(), defaultValue.textValue()))
+						{
+							combo.setSelection(new StructuredSelection(node));
+							break;
+						}
+					}
```

Comment:
Why don't we use the API ``values.get(defaultValue.textValue()) != `` instead of iterating through all the values in the array.